doc/smart-switch/smart-switch-database-architecture/smart-switch-database-design.md

```diff
@@ -0,0 +1,262 @@
+},
+"DPU_COUNTERS_DB": {
+    "id": 18,
+    "separator": ":",
+    "instance": "redis"
+}
+```
+
+#### Database flow
+
+This section outlines critical workflows interacting with the DPU overlay database.
+
+- Update Overlay Objects via GNMI:
+
+Communication with the SWSS of the DPU occurs through GNMI, leveraging ZMQ. Simultaneously, an asynchronous insertion of the object backup is made to the DPU_APPL_DB. This backup mechanism serves purposes such as debugging, migration, and future considerations.
+
+- Update Object Status:
+
+The SWSS of the DPU takes a proactive role in updating the DPU_APPL_STATE_DB and DPU_STATE_DB when corresponding objects undergo updates. This update can be triggered either by GNMI message commands or internal service logic.
+
+- Update Counters and Meters:
+
+Flex counter management in Syncd of the DPU handles the update of counters and meters for overlay objects. Traditional counters are also managed through this mechanism.
+
+These workflows ensure an interaction between the DPU overlay database and various components within the Smart Switch. The DPUs access their respective database instances via the IP address of the midplane bridge and the assigned TCP port. Concurrently, GNMI accesses these instances through the Unix domain socket
+
+### High-Level Design 
+
+### SAI API 
+
+N/A
+
+### Configuration and management 
+
+No new CLI commands are required
+
+#### CLI/YANG model Enhancements 
+
+
+``` yang
+    container sonic-feature {
+        container FEATURE {
+                leaf has_per_dpu_scope {
+                    description "This configuration identicates there will only one service
+                                spawned per DPU";
+                    type feature-scope-status;
+                    default "false";
+                }
+        }
+    }
+```
+
+#### Config DB Enhancements  
+
+Refer section: [Database services](#database-services)
+
+        
+### Warmboot and Fastboot Design Impact
+
+N/A
+
+### Memory Consumption
+
+#### DPU_APPL_DB
+
+The estimated memory consumption for the Smart Switch database is calculated based on entry sizes sourced from the [sonic-dash-api](https://github.com/sonic-net/sonic-dash-api) repository and entry numbers derived from the [DASH high-level design scaling requirements](https://github.com/sonic-net/DASH/blob/main/documentation/general/dash-sonic-hld.md#14-scaling-requirements).
+
+The following tables comprises two parts: Global tables and per ENI tables. Notably, when calculating the total size per card, the memory consumption of per ENI tables is adjusted by multiplying it by the exact number of ENIs.
+
+- Global Tables
+
+| Table name              | Entry size (bytes) | No. of entries in the Table | Total size per card (KB) |
+| ----------------------- | ------------------ | --------------------------- | ------------------------ |
+| DASH_VNET_TABLE         | 448                | 1,024                       | 448                      |
```

Comment:
the size of tables might change overtime, so should we put a specific commit id to make the reference more explicit?

Reply:
Done